pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -138,18 +139,23 @@ func EnforceHostNetwork() bool {
 }
 
 func SetRevisionHistoryLimit(data map[string]string) {
-	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultValue)
-	if strval != "" {
-		numval, err := strconv.ParseInt(strval, 10, 32)
-		if err != nil {
-			logger.Warningf("failed to parse value %q for %q. assuming default value.", strval, revisionHistoryLimitSettingName)
-			revisionHistoryLimit = nil
-			return
-
-		}
-		limit := int32(numval)
-		revisionHistoryLimit = &limit
+	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultStr)
+	var limit int32
+	if strval == "" {
+		//limit = revisionHistoryLimitDefault
+		logger.Warningf("not parsing empty string to int for %q. assuming default value.", revisionHistoryLimitSettingName)
+		revisionHistoryLimit = nil
+		return
+	}
+	numval, err := strconv.ParseInt(strval, 10, 32)
```

Comment:
Why `ParseInt()` instead of the less complex `Atoi()`? We have to use `int32()` to cast the variable type either way, so we might as well use a less complex function to read.

Reply:
good idea. done.

Reply:
After changing it, I notice that using `Atoi` leads to a lint ci-check error:

```
G109: Potential Integer overflow made by strconv.Atoi result conversion to int16/32 (gosec)
```
A bit of web search tells me that `ParseInt` is the  preferred way of converting a string to an integer of particular bitsize and that `Atoi`is anyway a wrapper to ``ParseInt`. I am therefore considering to revert back to using `ParseInt`.

Reply:
done, fwiw ...

Reply:
Sounds good to avoid possible int overflow errors